traffic_ops/traffic_ops_golang/server/servers_assignment.go

```diff
@@ -81,6 +89,31 @@ func AssignDeliveryServicesToServerHandler(w http.ResponseWriter, r *http.Reques
 	api.WriteRespAlertObj(w, r, tc.SuccessLevel, "successfully assigned dses to server", tc.AssignedDsResponse{server, assignedDSes, replace})
 }
 
+// ValidateDSCapabilities checks that the server meets the requirements of each delivery service to be assigned.
+func ValidateDSCapabilities(dsIDs []int, serverName string, tx *sql.Tx) error {
```

Comment:
this need to differentiate between storage errors from `GetServerCapabilitiesFromName` and not meeting the delivery service req capabilities.

This is because the return back to the user differs. The first should be an internal server error and the latter needs to be bad request.

Right now getting internal server error on both.